### PR TITLE
DIRECTORY TREE GENERATOR

### LINKED ISSUE #530: Directory Tree Generator
# Aim

Directory Tree Generator program will be useful for visualizing the relationship between files and directories and making their positioning easy.

# Details

It will use Docopt and Argparse for the project

I am gssoc participant and please assign me the issue

### PR DESCRIPTION
**Pull request by vikashkumar2020**
# Description

A Script useful for visualizing the relationship between files and directories and making their positioning easy.
- This can be used for implementations as one of the features.

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by santushtisharma10** [comment]
LGTM
@kaustubhgupta @antrikshmisri

**Review by santushtisharma10 — APPROVED** [review]

File: Directory Tree Generator/tree.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    if root is not None:
+        path=os.path.join(root, path)
+    result=os.path.basename(path)
+    if os.path.islink(path):
+        realpath=os.readlink(path)
+        result= '%s -> %s' % (os.path.basename(path), realpath)
+    return result
+
+def ptree(startpath, depth=-1):
+    prefix=0
```
Comment (antrikshmisri) [inline Directory Tree Generator/tree.py:15]:
```suggestion
    prefix=0
    assert os.path.isdir(startpath),"Directory not valid"
```

**Review by antrikshmisri — CHANGES_REQUESTED** [review]
@vikashkumar2020 Everything seems good. Just add this line to handle any invalid directories

File: Directory Tree Generator/tree.py (hunk 1 above)
Comment (vikashkumar2020) [reply Directory Tree Generator/tree.py:15]: changes are done and you can review it now

**Review by antrikshmisri — APPROVED** [review]
LGTM 🚀

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: Directory Tree Generator/tree.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        print('{}{}/'.format(indent, realname(root)))
+       
+        for d in dirs:
+            if os.path.islink(os.path.join(root, d)):
+                print('{}{}'.format(subindent, realname(d, root=root)))
+        for f in files:
+            print('{}{}'.format(subindent, realname(f, root=root)))
+
+if __name__ == '__main__':
+    import argparse
```
[the remark this answers is not in the document]
Comment (vikashkumar2020) [reply Directory Tree Generator/tree.py]: All the changes are done, please review

**Review by kaustubhgupta — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM..🚩

**Comment by avinashkranjan** [comment]
@all-contributors please add @vikashkumar2020 for code and documentation

**Pull request merged**